mlflow/utils/search_utils/parse.py

```diff
@@ -0,0 +1,231 @@
+import sqlparse
+from sqlparse.sql import Identifier as SqlIdentifier, Token as SqlToken, \
+    Comparison as SqlComparison, Statement as SqlStatement
+from sqlparse.tokens import Token as SqlTokenType
+
+from mlflow.utils.search_utils.models import Comparison, ComparisonOperator, KeyType
+from mlflow.exceptions import MlflowException
+from mlflow.protos.databricks_pb2 import INVALID_PARAMETER_VALUE
+
+
+KEY_TYPE_FROM_IDENTIFIER = {
+    "metric": KeyType.METRIC,
+    "metrics": KeyType.METRIC,
+    "param": KeyType.PARAM,
+    "params": KeyType.PARAM,
+    "parameter": KeyType.PARAM,
+    "tag": KeyType.TAG,
+    "tags": KeyType.TAG
+}
+
+STRING_VALUE_SQL_TYPES = {SqlTokenType.Literal.String.Single}
+NUMERIC_VALUE_SQL_TYPES = {SqlTokenType.Literal.Number.Integer, SqlTokenType.Literal.Number.Float}
+
+INVALID_IDENTIFIER_TPL = (
```

Comment:
Micronit: could we consistently pluralize (or not pluralize) metrics, params, and tags here? e.g. either use `"metrics.<key>", "params.<key>", "tags.<key>"` or "metric.<key>", "param.<key>", "tag.<key>"`? I personally prefer the plural form but singular could be ok too.